lib/gems/pending/util/MiqSshUtil.rb

```diff
@@ -40,13 +40,23 @@ def initialize(host, user, password, options = {})
 
   def get_file(from, to)
     run_session do |ssh|
-      $log.debug "MiqSshUtil::get_file - Copying file #{@host}:#{from} to #{to}." if $log
+      $log&.debug("MiqSshUtil::get_file - Copying file #{@host}:#{from} to #{to}.")
       data = ssh.sftp.download!(from, to)
-      $log.debug "MiqSshUtil::get_file - Copying of #{@host}:#{from} to #{to}, complete." if $log
+      $log&.debug("MiqSshUtil::get_file - Copying of #{@host}:#{from} to #{to}, complete.")
       return data
     end
   end
 
+  def put_file(to, content = nil, path = nil)
+    raise "Need to provide either content or path" if content.nil? && path.nil?
+    run_session do |ssh|
+      content ||= File.open(path, 'rb') { |f| f.read }
```

Comment:
How is this different from `IO.read(path)` ?

Just a little curious.

(sorry, drive by PR comment)

Reply:
Good question. It's a solution from stackoverflow... So, I'm ok with whatever is the preferred way.

Reply:
Reading further, it seems that `IO.binread` has been introduced in Ruby 1.9.1, and should be preferred. So, I updated my code. Thanks for pointing this out.